backend/typescript/services/implementations/ReviewService.ts

```diff
@@ -0,0 +1,174 @@
+import { Sequelize } from "sequelize-typescript";
+import PgReview from "../../models/review.model";
+import PgBook from "../../models/book.model";
+import PgTag from "../../models/tag.model";
+import PgSeries from "../../models/series.model";
+import PgAuthor from "../../models/author.model";
+import PgPublisher from "../../models/publisher.model";
+import logger from "../../utilities/logger";
+import { sequelize } from "../../umzug";
+import testSequelize from "../../testUtils/testDb";
+import {
+  ReviewRequestDTO,
+  IReviewService,
+  ReviewResponseDTO,
+  Book,
+  Author,
+  Publisher,
+  Tag,
+} from "../interfaces/IReviewService";
+
+const Logger = logger(__filename);
+
+// Delete: can delete the book, book_author, and book_publisher
+// Get: return review - but it contains all other info
+
+class ReviewService implements IReviewService {
+  db: Sequelize;
+
+  constructor(isTest = false) {
+    if (isTest) {
+      this.db = testSequelize;
+    } else {
+      this.db = sequelize;
+    }
+  }
+
+  /* eslint-disable class-methods-use-this */
+  async createReview(review: ReviewRequestDTO): Promise<ReviewResponseDTO> {
+    let publisher: PgPublisher;
+    let result: ReviewResponseDTO;
+
+    try {
+      result = await this.db.transaction(async (t) => {
+        const newReview = await PgReview.create(
+          {
+            body: review.body,
+            cover_images: review.coverImages,
+            byline: review.byline,
+            featured: review.featured,
+            published_at: review.publishedAt
+              ? new Date(review.publishedAt * 1000)
+              : null,
+          },
+          { transaction: t },
+        );
+
+        const pgTagsRet: PgTag[] = await Promise.all(
+          review.tags.map(async (reviewTag) => {
+            const tag = await PgTag.findOrCreate({
+              where: { name: reviewTag.name },
+              transaction: t,
+            }).then((data) => data[0]);
+            await newReview.$add("tags", tag, { transaction: t });
+            return tag;
```

Comment:
Here, rather than returning tag, could you return
`{
  name: tag.name
}`
And then this would allow you to directly define tagsRet: Tag[] in line 57 rather than needing to build it in the next few lines